omod-common/src/main/java/org/openmrs/module/webservices/rest/web/ConversionUtil.java

```diff
@@ -238,15 +240,15 @@ public static Object convert(Object object, Type toType) throws ConversionExcept
 
 
 			if (toClass.isAssignableFrom(Date.class)) {
-				ParseException pex = null;
+				IllegalArgumentException pex = null;
 				String[] supportedFormats = {"yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS",
 						"yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ssXXX", "yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm:ss", "yyyy-MM-dd"};
 				for (int i = 0; i < supportedFormats.length; i++) {
 					try {
-						Date date = new SimpleDateFormat(supportedFormats[i]).parse(string);
+						Date date = DateTime.parse(string, DateTimeFormat.forPattern(supportedFormats[i])).toDate();

```

Comment:
@lluismf As for `Date date = DateTime.parse(string).toDate();`, the last test, _convert_shouldReturnEqualsDateFromString()_ fails. The error thrown is `org.openmrs.module.webservices.rest.web.response.ConversionException: Error converting date - correct format (ISO8601 Long): yyyy-MM-dd'T'HH:mm:ss.SSSZ`
<br>
Taking the workflow and everything related to the tests in consideration, I felt that the way I have currently implemented the parsing seems to work out fine :)